src/cmr/sizing/core.clj

```diff
@@ -22,32 +22,59 @@
 ;;; Note that all internal operations for estimating size are performed
 ;;; assuming bytes as the units.
 
-(defn get-measurement
+(defn- get-measurement
   [variable]
   (let [dims (get-in variable [:umm :Dimensions])
         total-dimensionality (reduce * (map :Size dims))
         data-type (get-in variable [:umm :DataType])]
     (* total-dimensionality
        (util/data-type->bytes data-type))))
 
+(defn- get-avg-gran-size
+  "Gets SizeEstimation value for AverageSizeOfGranulesSampled and parses it to a number."
+  [vars]
+  (let [value (get-in (first vars) [:umm :SizeEstimation :AverageSizeOfGranulesSampled])]
+    (if (string? value)
+      (read-string value)
+      value)))
+
+(defn- get-avg-compression-rate
+  "Gets SizeEstimation value for AvgCompressionRateASCII and parses it to a number."
+  [vars]
+  (let [value (get-in (first vars) [:umm :SizeEstimation :AvgCompressionRateASCII])]
+    (if (string? value)
+      (read-string value)
+      value)))
+
 (defn estimate-dods-size
   [granule-count variables]
   (let [compression 1
         metadata 0 ; included here for symmetry
         measurements (reduce + (map get-measurement variables))]
     (+ (* granule-count compression measurements) metadata)))
 
-(defn estimate-netcdf3-size
+(defn- estimate-netcdf3-size
   [granule-count variables metadata]
   (let [compression 1
         measurements (reduce + (map get-measurement variables))]
     (+ (* granule-count compression measurements) metadata)))
 
+(defn- estimate-ascii-size
+  "Calculates the estimated size for ASCII format."
+  [granule-count variables params]
+  (let [metadata 0
```

Comment:
Bob clarified that all the values are actually variable specific. So what we'll need to do is more like:
```
(reduce (fn [total-estimate var]
          (let [avg-variable-size (get-avg-variable-size var)
                 ...   
                 var-estimate ((* edsc-size-estimate (+ (* (/ (/ edsc-size-estimate granule-count) avg-gran-size) avg-compression-rate) metadata)]
            (+ total-estimate var-estimate)
         0
         vars)
```